geoips_clavrx/plugins/modules/readers/clavrx_hdf4.py

```diff
@@ -41,110 +67,210 @@ def parse_metadata(metadatadict):
+def year_day_hours_to_datetime(year, day, time):
+    """Convert year, day, and time to a datetime object.
+
+    Parameters
+    ----------
+    year : int
+        The year.
+    day : int
+        Day of the year (1-366).
+    time : float
+        Time in hours.
+
+    Returns
+    -------
+    datetime
+        T
+    """
+    return datetime(year, 1, 1) + timedelta(days=day, hours=time)
 
 
 def read_cloudprops(fname, chans=None, metadata_only=False):
-    """Read CLAVR-x Cloud Properties Data."""
-    try:
-        data = SD(fname, SDC.READ)  # read in all data fields
-    except HDF4Error:
-        LOG.info("wrong input hdf file %s", fname)
-        raise
-
-    # selected cloud variables
-    # definiation of variables
-    # cira_out = {'latitude':latitude, 'longitude':longitude,
-    #     'cloud_type':cld_type,
-    #     'cloud_mask':cld_mask,
-    #     'cloud_phase':cld_phase,
-    #     'cloud_fraction',cld_fract,
-    #     'cld_height_acha':cld_hgt,
-    #     'cld_height_base_acha':cld_hgt_base,
-    #     'cld_height_top_acha':cld_hgt_top,
-    #     'cld_temp_acha':cld_temp,
-    #     'cloud_water_path':cwp,
-    #     'cld_opd_acha':cld_opd,
-    #     'cld_reff_acha':cld_reff,
-    #     'temp_3_75um_nom',tb_3p75,
-    #     'temp_11_0um_nom':tb_11p0,
-    #     'solar_zenith_angle':sza}
-
-    # Note:  Values of the attribute 'valid_range' for cloud_type, cloud_mask,
-    #        cloud_phase(?) are not valid.
-    #        They should be specified with their definitions.
+    """Read CLAVR-x Cloud Properties Data.
 
-    if chans is None:
-        vars_sel = sorted(data.datasets().keys())
-    elif chans:
-        vars_sel = chans
-    else:
-        metadata_only = True
+    Parameters
+    ----------
+    fname : str
+        Path to the HDF4 file.
+    chans : list of str, optional
+        List of variables to read. If None, all variables are read.
+    metadata_only : bool, optional
+        If True, only read metadata.
 
-    # process of all variables
-    xarrays = {}
+    Returns
+    -------
+    Dict("DATA": xarray.Dataset)
+
+    Notes
+    -----
+    Dataset contains cloud observation data with variables:
+        - 'latitude': Latitude.
+        - 'longitude': Longitude.
+        - 'cloud_type': Cloud type.
+        - 'cloud_mask': Cloud mask.
+        - 'cloud_phase': Cloud phase.
+        - 'cloud_fraction': Cloud fraction.
+        - 'cld_height_acha': Cloud height.
+        - 'cld_height_base_acha': Base cloud height.
+        - 'cld_height_top_acha': Top cloud height.
+        - 'cld_temp_acha': Cloud temperature.
+        - 'cloud_water_path': Cloud water path.
+        - 'cld_opd_acha': Cloud optical depth.
+        - 'cld_reff_acha': Effective cloud particle radius.
+        - 'temp_3_75um_nom': Brightness temperature at 3.75 µm.
+        - 'temp_11_0um_nom': Brightness temperature at 11.0 µm.
+        - 'solar_zenith_angle': Solar zenith angle.
+    """
```

Comment:
Just a note, but this set of variables can change based on how CLAVR-x is configured. I'm not sure that this is the full set of variables, just what happens to be available in the files we're currently using. Even if it is the full current set, new variables sometimes get added to CLAVR-x.

I'm not sure what I'm suggesting. Just pointing out that this may or may not be an exhaustive list.